driver-core/src/main/java/com/datastax/driver/core/LatencyTracker.java

```diff
@@ -31,15 +31,21 @@
      * the duration of that operation.
      * <p>
      * Note that there is no guarantee that this method won't be called
-     * concurrently by multiple thread, so implementations should synchronize
+     * concurrently by multiple threads, so implementations should synchronize
      * internally if need be.
      *
-     * @param host the Cassandra host on which a request has been performed.
-     * @param newLatencyNanos the latency in nanoseconds of the operation. This
-     * latency corresponds to the time elapsed between when the query was send
-     * to {@code host} and when the response was received by the driver (or the
-     * operation timed out, in which {@code newLatencyNanos} will approximately
-     * be the timeout value).
+     * @param host The Cassandra host on which a request has been performed.
+     *             This parameter is never {@code null}.
+     * @param statement The {@link com.datastax.driver.core.Statement} that has been executed.
+     *                  This parameter is never {@code null}.
```

Comment:
We can now guarantee that the statement is never null (it is never null in RequestHandler where the trackers are called). It is never Statement.DEFAULT neither (see below).